tests/AMSlib/wf/policy.cpp

```diff
@@ -0,0 +1,129 @@
+#include "wf/policy.hpp"
+
+#include <catch2/catch_test_macros.hpp>
+#include <memory>
+#include <string>
+#include <type_traits>
+
+#include "ml/Model.hpp"
+#include "wf/action.hpp"
+#include "wf/eval_context.hpp"
+#include "wf/layout_transform.hpp"
+#include "wf/pipeline.hpp"
+
+namespace ams
+{
+
+namespace
+{
+
+class IncAction final : public Action
+{
+public:
+  const char* name() const noexcept override { return "IncAction"; }
```

Comment:
In `Action`, `Policy`, basically every abstract layers with a method `name()`, we could have something like that by default:
```c++
    virtual const char* name() const {
        // +1 to remove the first character which is the number of letters in the name (C++ default)
        return typeid(*this).name()+1;
    }
```
Users could override it but at least all the tests etc would be simpler.